src/script/bitcoinconsensus.h

```diff
@@ -54,7 +56,7 @@ enum
 /// txTo correctly spends the scriptPubKey pointed to by scriptPubKey under
 /// the additional constraints specified by flags.
 /// If not NULL, err will contain an error/success code for the operation
-EXPORT_SYMBOL int bitcoinconsensus_verify_script(const unsigned char *scriptPubKey, unsigned int scriptPubKeyLen,
+EXPORT_SYMBOL int bitcoinconsensus_verify_script(const unsigned char *scriptPubKey, unsigned int scriptPubKeyLen, uint64_t amount,
```

Comment:
Since the amount to be signed is part of the txFrom but not part of the txTo, it seems necessary to modify the interface if we want signatures to cover input amounts. I guess that means increasing the version.

About having 2 functions...it seems that whether the input amount will be hashed for signing or not depends on sigversion, which is contained which can be deduced from the scriptPubKey, making it unnecessary to have to functions and have the caller decide which one has to use in each case. Am I missing something?